pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -228,12 +229,21 @@ func manageKubeAPIServerConfig(ctx context.Context, client coreclientv1.ConfigMa
 	configOverrides := bindata.MustAsset("assets/config/config-overrides.yaml")
 	specialMergeRules := map[string]resourcemerge.MergeFunc{}
 
+	// Guarantee the authorization-mode will be present in the base config, regardless of whether the observer is running
+	authModeOverride := map[string]interface{}{}
```

Comment:
nit: technically, this is our default configuration, so I would rename it to `defaultAuthModes`.